ush/ioda/bufr2ioda/bufr2ioda_adpupa.py

```diff
@@ -191,6 +206,7 @@ def bufr_to_ioda(config, logger):
     dewpointTemperatureOE = np.float32(np.ma.masked_array(np.full((len(tmdp)), 0.0)))
     windEastwardOE = np.float32(np.ma.masked_array(np.full((len(wspd)), 0.0)))
     windNorthwardOE = np.float32(np.ma.masked_array(np.full((len(wspd)), 0.0)))
+    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))
```

Comment:
@PraveenKumar-NOAA , `bufr2ioda_adpupa.py` aborted on WCOSS2 (Cactus) on this line with the following traceback
```
  File "/lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/ensda_obs/sorc/gdas.cd/ush/ioda/bufr2ioda/bufr2ioda_adpupa.py", line 209, in bufr_to_ioda
    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))
UnboundLocalError: local variable 'qob' referenced before assignment
```

This is a valid error.  `qob` is not assigned values until line 227
```
    qob = Compute_SpecificHumidity_from_dewPoint_and_Pressure(tmdp, prlc)
    logger.debug(f'   qob min/max = {qob.min()} {qob.max()}')
```

As a test I moved line 209 after line 227
```
    qob = Compute_SpecificHumidity_from_dewPoint_and_Pressure(tmdp, prlc)
    logger.debug(f'   qob min/max = {qob.min()} {qob.max()}')
    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))        
```
This worked on Cactus.

Is this an acceptable solution or is there a better solution?

Reply:
@RussTreadon-NOAA this is acceptable, thanks.

Reply:
Thank you @PraveenKumar-NOAA for the confirmation.  I will add this fix to GDASApp PR #1252